modules/apiserversource-kn.adoc

```diff
@@ -0,0 +1,143 @@
+// Module included in the following assemblies:
+//
+// serverless/knative_eventing/serverless-kn-source.adoc
+
+[id="apiserversource-kn_context"]
+= Using the ApiServerSource with the Knative CLI (`kn`)
+
+This guide describes the steps required to create, manage and delete an ApiServerSource using `kn` commands.
+
+.Prerequisites
+
+* You will need to have a Knative Serving and Eventing installation.
+* You will need to have created the `default` broker in the same namespace that the ApiServerSource will be installed in.
+* You will need to have the `kn` CLI installed.
+
+.Procedure
+
+. Create a service account, cluster role and cluster binding for the ApiServerSource.
++
+You can do this by creating a file named `authentication.yaml` and copying the following sample code into it:
++
+----
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: events-sa
+  namespace: default <1>
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: event-watcher
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - events
+    verbs:
+      - get
+      - list
+      - watch
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: k8s-ra-event-watcher
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: event-watcher
+subjects:
+  - kind: ServiceAccount
+    name: events-sa
+    namespace: default <1>
+----
++
+<1> Adjust this namespace to the namespace that you have selected for installing ApiServerSource.
++
+[NOTE]
+====
+If you want to re-use an existing service account with the appropriate permissions, you must modify the `authentication.yaml` for that service account.
+====
++
+Create the service account, cluster role and cluster binding by entering the following command:
++
+----
+$ oc apply --filename authentication.yaml
+----
+
+. Create an ApiServerSource event source by entering the following command:
++
+----
+$ kn source apiserver create testevents --sink broker:default --resource "event:v1" --service-account events-sa
+----
+
+. To check that the ApiServerSource is set up correctly, create a Knative service that dumps incoming messages to its log by entering the following command:
++
+----
+$ kn service create event-display --image quay.io/openshift-knative/knative-eventing-sources-event-display
+----
+
+. Create a trigger from the `default` broker to filter events to the service created in the previous step by entering the following `kn` command:
++
+----
+$ kn trigger create event-display-trigger --broker default --sink svc:event-display \ source=dev.knative.samples/testevents \
+----
+
+. Create events by launching a pod in the default namespace.
```

Comment:
If we have a one-liner for the lazy readers to do this, could we add it here?  (e.g. one of those that "oc new-project" suggests to use to start... 

e.g. oc create deployment hello-node --image=gcr.io/hello-minikube-zero-install/hello-node

Reply:
Added this as a placeholder, but is this image OK to use? It seems off that we use quay for the others and gcr for this - also it refers to minikube?

Reply:
In my demos I'm using a 

```
oc run --image rhsummit2020/random:1.0 --generator=run-pod/v1 random
```

however, you could any image here (like @maschmid suggests). This create a bare pod that can be easily deleted directly.

Reply:
Not sure what a good quay.io sample image could be ?

Reply:
With OCP 4.4 oc,  the "oc new-project" suggests these

> Now using project "foobar2" on server "https://api.it-31-internal-tests.openshift-aws.rhocf-dev.net:6443".
> 
> You can add applications to this project with the 'new-app' command. For example, try:
> 
>     oc new-app ruby~https://github.com/sclorg/ruby-ex.git
> 
> to build a new example application in Python. Or use kubectl to deploy a simple Kubernetes application:
> 
>     kubectl create deployment hello-node --image=gcr.io/hello-minikube-zero-install/hello-node
> 
> 

that's why I suggested gcr.io/hello-minikube-zero-install/hello-node

Reply:
I got you, that makes sense! Thanks.